packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -3472,23 +3478,145 @@ const TextStyle _messageStyle = TextStyle(
+  static const double _kExitWidgetSelectionButtonMargin = 10.0;
+
+  final GlobalKey _exitWidgetSelectionButtonKey = GlobalKey(
+    debugLabel: 'Exit Widget Selection button',
+  );
+
+  String? _tooltipMessage;
+
+  bool _leftAligned = true;
+
+  @override
+  Widget build(BuildContext context) {
+    final Widget? moveExitWidgetSelectionButton =
+        widget.moveExitWidgetSelectionButtonBuilder != null
+            ? Padding(
+                padding: _leftAligned
+                    ? const EdgeInsets.only(
+                        left: _kExitWidgetSelectionButtonPadding,
+                      )
+                    : const EdgeInsets.only(
+                        right: _kExitWidgetSelectionButtonPadding,
+                      ),
+                child: _TooltipGestureDetector(
+                  button: widget.moveExitWidgetSelectionButtonBuilder!(
+                    context,
+                    onPressed: () {
+                      _changeButtonGroupAlignment();
+                      _onTooltipHidden();
+                    },
+                    isLeftAligned: _leftAligned,
+                  ),
+                  onTooltipVisible: () {
+                    _changeTooltipMessage(
+                      'Move to the ${_leftAligned ? 'right' : 'left'}',
+                    );
+                  },
+                  onTooltipHidden: _onTooltipHidden,
+                ),
+              )
+            : null;
+
+    final Widget buttonGroup = Stack(
+      alignment: AlignmentDirectional.topCenter,
+      children: <Widget>[
+        CustomPaint(
+          painter: _ExitWidgetSelectionTooltipPainter(
+            tooltipMessage: _tooltipMessage,
+            buttonKey: _exitWidgetSelectionButtonKey,
+            isLeftAligned: _leftAligned,
+          ),
+        ),
+        Row(
+          children: <Widget>[
+            if (!_leftAligned && moveExitWidgetSelectionButton != null)
+              moveExitWidgetSelectionButton,
+            _TooltipGestureDetector(
+              button: widget.exitWidgetSelectionButtonBuilder(
+                context,
+                onPressed: _exitWidgetSelectionMode,
+                key: _exitWidgetSelectionButtonKey,
+              ),
+              onTooltipVisible: () {
+                _changeTooltipMessage('Exit Select Widget mode');
+              },
+              onTooltipHidden: _onTooltipHidden,
+            ),
+            if (_leftAligned && moveExitWidgetSelectionButton != null)
+              moveExitWidgetSelectionButton,
+          ],
+        ),
+      ],
+    );
+
+    return _leftAligned
+        ? Positioned(
+            left: _kExitWidgetSelectionButtonMargin,
+            bottom: _kExitWidgetSelectionButtonMargin,
+            child: buttonGroup,
+          )
+        : Positioned(
+            right: _kExitWidgetSelectionButtonMargin,
+            bottom: _kExitWidgetSelectionButtonMargin,
+            child: buttonGroup,
+          );
```

Comment:
nit: this can be condensed
```
Positioned(
  left: _leftAligned ? _kExitWidgetSelectionButtonMargin :  null,
  right: _leftAligned ? null : _kExitWidgetSelectionButtonMargin,
  bottom: _kExitWidgetSelectionButtonMargin,
  child: buttonGroup,
)
```

Reply:
Thanks, done!